src/mixins/java/org/spongepowered/common/mixin/api/minecraft/resources/ResourceLocationMixin_API.java

```diff
@@ -45,15 +46,20 @@ public abstract class ResourceLocationMixin_API {
     @Shadow public abstract String shadow$getPath();
     // @formatter:on
 
+    /**
+     * @author MrHell228 - October 6th, 2025
+     * @reason Both Key#compareTo and ResourceLocation#compareTo should use the same way to compare values.
+     */
+    @Overwrite
+    public int compareTo(final ResourceLocation location) {
+        return ((Key) this).compareTo((Key) (Object) location);
```

Comment:
```suggestion
    public int compareTo(final Object obj) {
        return ((Key) this).compareTo((Key) obj);
```

After looking at the implementations of `ResourceLocation#compareTo(ResourceLocation)` and `Key#compareTo(Key)`, they seem completely equivalent to me. I don't think we need to overwrite one or the other. What I'm more worried about is the synthetic method `ResourceLocation#compareTo(Object)` which is casting and delegating to `ResourceLocation#compareTo(ResourceLocation)`. I think this one should be overwritten to delegate to `Key#compareTo(Key)` instead.

Reply:
Makes sense. Didn't know it works via synthetic method. I guess now `ResourceLocation#compareTo(ResourceLocation)` should not be reachable from the API and no CCE should be thrown if anyone implements `Key` on their own (or if `KeyMixin` is removed which would result in Adventure `KeyImpl` being used).